### PR TITLE
optimize bmp180 sensor adapt to uorb.

### PR DESCRIPTION
**Pull request by Otpvondoiats**
## Summary
provides work in character device mode and UORB communication mode.

Roll back the bmp180 code and create a new bmp180_uorb file.
Use the macro CONFIG_SENSORS_BMP180_UORB to enable the uorb feature.

## Impact
None
## Testing
ubuntu 20.04
I2C/SIM 
UORB and character device mode

### CODE REVIEW

File: drivers/sensors/bmp180_uorb.c [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    {
+      snerr("ERROR: Failed to register driver: %d\n", ret);
+      kmm_free(priv);
+    }
+
+  sninfo("BMP180 driver loaded successfully!\n");
+  return ret;
+}
+
+#endif /* CONFIG_I2C && CONFIG_SENSORS_BMP180 */
```
Comment (xiaoxiang781216) [inline drivers/sensors/bmp180_uorb.c:276]: add blank line

Comment (Otpvondoiats) [reply drivers/sensors/bmp180_uorb.c]: Done

File: boards/arm/stm32l4/nucleo-l476rg/src/stm32_bmp180.c [hunk 2]
```diff
@@ -81,4 +81,4 @@ int stm32_bmp180initialize(const char *devpath)
   return ret;
 }
 
-#endif /* CONFIG_I2C && CONFIG_SENSORS_BMP180 */
+#endif /* CONFIG_I2C && CONFIG_SENSORS_BMP180 */
```
[the remark this answers is not in the document]
Comment (Otpvondoiats) [reply boards/arm/stm32l4/nucleo-l476rg/src/stm32_bmp180.c]: Done

**Comment by xiaoxiang781216** [comment]
@raiden00pl could you review bmp180 change?

**Review by raiden00pl — APPROVED** [review]
LGTM. Mac failure not related

**Pull request merged**